WordPress/Classes/ViewRelated/Comments/Controllers/Edit/CommentEditViewModel.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import CoreData
+
+final class CommentEditViewModel {
+    let suggestionsViewModel: SuggestionsListViewModel?
+
+    private let comment: Comment
+    private let siteID: NSNumber
+    private let context =  ContextManager.shared.mainContext
+
+    var isGutenbergEnabled: Bool {
+        FeatureFlag.readerGutenbergCommentComposer.enabled
+    }
+
+    /// Edit an existing comment.
+    init(comment: Comment) {
+        self.comment = comment
+        self.siteID = comment.associatedSiteID ?? 0
+        wpAssert(siteID != 0, "missing required parameter siteID")
+
+        self.suggestionsViewModel = SuggestionsListViewModel.make(siteID: siteID)
+        self.suggestionsViewModel?.enableProminentSuggestions(
+            postAuthorID: comment.post?.authorID,
+            commentAuthorID: comment.commentID as NSNumber
+        )
+    }
+
+    var originalContent: String {
+        comment.rawContent
+    }
+
+    @MainActor
+    func save(content: String) async throws {
+        let commentID = comment.commentID as NSNumber
+        let service = CommentService(coreDataStack: ContextManager.shared)
+
+        let remoteComment = try await withUnsafeThrowingContinuation { continuation in
```

Comment:
What are the benefits of using `withUnsafeThrowingContinuation` vs `withCheckedThrowingContinuation` in this context?

Reply:
I don't think I've used `withCheckedThrowingContinuation` before because AFAIU it throws a `fatalError` if there is an error.

Reply:
On that aspect, I guess the "checked" version fails early, but who knows what it means if we are not "being extra careful".

> Unlike the “checked” continuation variant, the UnsafeContinuation does not detect or diagnose any kind of misuse, so you need to be extra careful to avoid calling resume twice or forgetting to call resume before letting go of the continuation object.